### DISCUSSION TITLE
String based operators vs Symbol based operators

### DISCUSSION THREAD

**Discussion by rosscoombes** [sequelize/sequelize#14044] [Help & Questions] [answered]
I've just run into this issue, in short, using the operator $like instead of [Op.like]
https://stackoverflow.com/questions/48465321/how-can-i-format-a-sequelize-filter-as-valid-json

I'm currently trying to allow a front end implementation to be able to pass a JSON string that I can use to filter the results of a sequelize query.  Perhaps I'm flawed in my approach but [Op.like] cannot be passed through JSON.stringify whereas $like can be.

The documentation says that the string based operators are deprecated and soon to be removed completely.  Given this is currently the simplest approach to do this, I would just like to propose that this "old" method is kept?

**Comment by ephys** [comment, marked as answer]
It's very unlikely that string operators will be kept since preventing the accidental injection of operators is the whole reason this change was made ([source from old v4 documentation](https://sequelize.org/v4/manual/tutorial/querying.html#operators-security))

The good news is that if you really need such a feature, it's easy enough to implement your own function to map an arbitrary key to one of our operator symbols: https://stackoverflow.com/questions/39123461/how-to-change-object-keys-in-deeply-nested-object-with-lodash

Update: string operators have been removed from Sequelize 7
